test/Microsoft.SqlTools.ServiceLayer.UnitTests/EditData/SessionTests.cs

```diff
@@ -606,6 +606,72 @@ public async Task RevertRowSuccess()
             Assert.That(s.EditCache.Keys, Has.No.Zero, "The edit cache should not contain a pending edit for the row");
         }
 
+        [Test]
+        public async Task RevertRowAfterAdding()
+        {
+            // Setup:
+            // ... Create a session with a proper query and metadata
+            EditSession s = await GetBasicSession();
+
+            EditRow[] rows = await s.GetRows(0, 10);
+
+            int defaultRowsLength = rows.Length;
+
+            // ... Create a row 
+            EditCreateRowResult result1 = s.CreateRow();
+
+            long result1Id = result1.NewRowId;
+
+            long addRowNextRowId = s.NextRowId;
+
+            rows = await s.GetRows(0, 10);
+
+            // Check that returned result of createRow matches NextRowId index.
+            Assert.AreEqual(result1Id, addRowNextRowId - 1);
+
+            // Check that row has been added.
+            Assert.AreEqual(rows.Length, defaultRowsLength + 1);
+
+            // Check that the nextRowId has a value that reflects the number of rows.
+            Assert.AreEqual(rows.Length, s.NextRowId);
+ 
+            // If: I revert the row that has a pending CreateRow (Such as when there's a row failure)
+            s.RevertRow(result1Id);
+
+            rows = await s.GetRows(0, 10);
+
+            // Check that the rowId has decremented after a revert.
```

Comment:
All of these comments on the asserts should also be reflected in the message for the assert - right now if any of these fail it's going to be difficult to figure out exactly where it failed without debugging in it. 

So for example this one should be something like 

`Assert.That(s.NextRowId, Is.EqualTo(addRowNextRowId - 1), "NextRowId should not be incremented after reverting a CreateRow edit");`

See https://github.com/microsoft/azuredatastudio-docs/blob/main/Guides/Testing-Best-Practices.md for some general guidelines to follow (including using the fluent assertion language NUnit provides)